dali/pipeline/operators/arithmetic/arithmetic_test.cc

```diff
@@ -0,0 +1,338 @@
+      std::make_tuple("mul", [](T l, T r) -> T { return l * r; }),
+      std::make_tuple("div", [](T l, T r) -> T { return l / r; }),
+      std::make_tuple("mod",
+                      std::is_integral<T>::value
+                      ? [](T l, T r) -> T { return std::fmod(l, r); }
+                      : [](T l, T r) -> T { return std::remainder(l, r); }),
+  };
+}
+
+// We need to pass a type name to macro TEST_P, so we use an alias
+// We will test for both backends with int32_t and float.
+using BinaryArithmeticOpCPUint32 = BinaryArithmeticOpTest<CPUBackend, int32_t>;
+using BinaryArithmeticOpCPUfloat = BinaryArithmeticOpTest<CPUBackend, float>;
+using BinaryArithmeticOpGPUint32 = BinaryArithmeticOpTest<GPUBackend, int32_t>;
+using BinaryArithmeticOpGPUfloat = BinaryArithmeticOpTest<GPUBackend, float>;
+
+// Create the tests
+TEST_P(BinaryArithmeticOpCPUint32, SimplePipeline) {
+  TestFunction();
+}
+TEST_P(BinaryArithmeticOpCPUfloat, SimplePipeline) {
+  TestFunction();
+}
+TEST_P(BinaryArithmeticOpGPUint32, SimplePipeline) {
+  TestFunction();
+}
+TEST_P(BinaryArithmeticOpGPUfloat, SimplePipeline) {
+  TestFunction();
+}
+
+// Pass the values to tests suites
+INSTANTIATE_TEST_SUITE_P(BinaryArithmeticOpsSuiteCPUint32, BinaryArithmeticOpCPUint32,
+                         ::testing::ValuesIn(getOpNameRef<int32_t>()));
+
+INSTANTIATE_TEST_SUITE_P(BinaryArithmeticOpsSuiteCPUfloat, BinaryArithmeticOpCPUfloat,
+                         ::testing::ValuesIn(getOpNameRef<float>()));
+
+INSTANTIATE_TEST_SUITE_P(BinaryArithmeticOpsSuiteGPUint32, BinaryArithmeticOpGPUint32,
+                         ::testing::ValuesIn(getOpNameRef<int32_t>()));
+
+INSTANTIATE_TEST_SUITE_P(BinaryArithmeticOpsSuiteGPUfloat, BinaryArithmeticOpGPUfloat,
+                         ::testing::ValuesIn(getOpNameRef<float>()));
+
+TEST(ArithmeticOps, GenericPipeline) {
+  constexpr int batch_size = 16;
+  constexpr int num_threads = 4;
+  constexpr int tensor_elements = 16;
+  Pipeline pipe(batch_size, num_threads, 0);
+
+  pipe.AddExternalInput("data0");
+  pipe.AddExternalInput("data1");
+
+  pipe.AddOperator(OpSpec("ArithmeticGenericOp")
+                       .AddArg("device", "cpu")
+                       .AddArg("expression_desc", "add(&0 &1)")
+                       .AddInput("data0", "cpu")
+                       .AddInput("data1", "cpu")
+                       .AddOutput("result", "cpu"),
+                   "arithm_cpu");
+
+  pipe.AddOperator(OpSpec("ArithmeticGenericOp")
+                       .AddArg("device", "gpu")
+                       .AddArg("expression_desc", "mul(&0 &1)")
+                       .AddInput("result", "gpu")
+                       .AddInput("data1", "gpu")
+                       .AddOutput("result2", "gpu"),
+                   "arithm_gpu");
+
+  vector<std::pair<string, string>> outputs = {{"result", "cpu"}, {"result2", "gpu"}};
+
+  pipe.Build(outputs);
+  pipe.SaveGraphToDotFile("test.dot");
```

Comment:
This is probably not used beyond debugging.

Reply:
Right. Removed